src/Cache/Engine/RedisEngine.php

```diff
@@ -135,6 +142,67 @@ protected function _connect(): bool
         return $return;
     }
 
+    /**
+     * Connects to a Redis server using a new connection.
+     *
+     * @param string $server Server to connect to.
+     * @param array $ssl SSL context options.
+     * @throws \RedisException
+     * @return bool True if Redis server was connected
+     */
+    protected function _connectTransient($server, array $ssl): bool
+    {
+        if (empty($ssl)) {
+            return $this->_Redis->connect(
+                $server,
+                (int)$this->_config['port'],
+                (int)$this->_config['timeout']
+            );
+        }
+
+        return $this->_Redis->connect(
+            $server,
+            (int)$this->_config['port'],
+            (int)$this->_config['timeout'],
+            null,
+            0,
+            0.0,
+            ['ssl' => $ssl]
```

Comment:
What will old versions of the redis extension do when they get and additional parameter? If they fail we will need to do version sniffing.